pkg/action/action.go

```diff
@@ -91,6 +93,76 @@ type Configuration struct {
 	mutex sync.Mutex
 }
 
+const (
+	// FilenameAnnotation is the annotation key used to store the original filename
+	// information in manifest annotations for post-rendering reconstruction.
+	FilenameAnnotation = "helm-postrender-filename"
+)
+
+// AnnotateAndMerge combines multiple YAML files into a single stream of documents,
+// adding filename annotations to each document for later reconstruction.
+func AnnotateAndMerge(files map[string]string) (string, error) {
```

Comment:
Similar to above, can we make this private(?)

Reply:
done